src/test/scala/vsys/blockchain/state/contract/channel/PaymentChannelContractDiffTest.scala

```diff
@@ -72,35 +84,357 @@ class PaymentChannelContractDiffTest extends PropSpec
+        blockDiffEi.explicitGet().txsDiff.portfolios.isEmpty shouldBe false
+        blockDiffEi.explicitGet().txsDiff.txStatus shouldBe TransactionStatus.Success
+      }
+    }
+  }
+
+  property("execute withdraw able to withdraw tokens from channel after channel expires") {
+    forAll(preconditionsAndPaymentChannelWithTokenFunctionTest) { case (genesis: GenesisTransaction, reg: RegisterContractTransaction,
+    reg2: RegisterContractTransaction, issue: ExecuteContractFunctionTransaction, deposit: ExecuteContractFunctionTransaction,
+    create: ExecuteContractFunctionTransaction, _, executeWithdraw: ExecuteContractFunctionTransaction, _, fee: Long, _) =>
+      assertDiffAndStateCorrectBlockTime(Seq(TestBlock.create(genesis.timestamp, Seq(genesis)), TestBlock.create(create.timestamp, Seq(reg, reg2, issue, deposit, create))),
+        TestBlock.create(create.timestamp + 1000000000001L, Seq(executeWithdraw))) { (blockDiff, newState) =>
+        blockDiff.txsDiff.txStatus shouldBe TransactionStatus.Success
+        val totalPortfolioDiff: Portfolio = Monoid.combineAll(blockDiff.txsDiff.portfolios.values)
+        totalPortfolioDiff.balance shouldBe - fee
+        totalPortfolioDiff.effectiveBalance shouldBe - fee
+
+        val master = reg.proofs.firstCurveProof.explicitGet().publicKey
+        val masterBytes = genesis.recipient.bytes.arr
+        val contractId = reg.contractId.bytes
+        val tokenContractId = reg2.contractId.bytes
+        val tokenId = tokenIdFromBytes(tokenContractId.arr, Ints.toByteArray(0)).explicitGet()
+
+        //stateMap keys
+        val channelId = DataEntry(Shorts.toByteArray(create.id.arr.length.toShort) ++ create.id.arr, DataType.ShortBytes).bytes
+        val balanceInContractKey = ByteStr(Bytes.concat(contractId.arr, Array(0.toByte), DataEntry(masterBytes, DataType.Address).bytes))
+        val channelCapacityInContractKey = ByteStr(Bytes.concat(contractId.arr,  Array(4.toByte), channelId))
+        val executedKey = ByteStr(Bytes.concat(contractId.arr,  Array(5.toByte), channelId))
+
+        val (_, masterTxs) = newState.accountTransactionIds(master, 7, 0)
+        masterTxs.size shouldBe 7 // genesis, reg, reg2, issue, deposit, create, executeWithdraw
+
+        //channel num info
+        newState.contractNumInfo(balanceInContractKey) shouldBe 10000L
+        newState.contractNumInfo(channelCapacityInContractKey) shouldBe 100L
+        newState.contractNumInfo(executedKey) shouldBe 100L
+
+        // token balance keys
+        val masterBalanceKey = ByteStr(Bytes.concat(tokenId.arr, masterBytes))
+        val contractBalanceKey = ByteStr(Bytes.concat(tokenId.arr, contractId.arr))
+
+        // VSYS balance
+        newState.balance(master.toAddress) shouldBe ENOUGH_AMT - 6 * fee - 2 * 10000000000L
+
+        // Token balance
+        newState.tokenAccountBalance(masterBalanceKey) shouldBe 100000L - 10000L
+        newState.tokenAccountBalance(contractBalanceKey) shouldBe 10000L
+      }
+    }
+  }
```

Comment:
One suggestion here: we may simplify the code by

 add some common generation function payment contract gen
```scala
// deposit vsys case
def func_name(deposit value, channel create amount, channel expired time): Gen[] = for {

} yield (two genesis, master, user, reg, deposit, create, ts, fee)

// deposit token case
def func_name(deposit value, channel create amount, channel expired time): Gen[] = for {

} yield (two genesis, master, user, regToken, issueToken, deposit, create, tokenId, ts, fee)
```